sync-core/src/main/java/com/cloudant/sync/query/QueryResult.java

```diff
@@ -44,23 +42,19 @@ public QueryResult(List<String> docIds,
                        long skip,
                        long limit,
                        UnindexedMatcher matcher) {
-        this.docIds = docIds;
+        this.docIds = matcher == null ? processDocIds(docIds, skip, limit) :
+                                        processDocIds(docIds, datastore, skip, limit, matcher);
```

Comment:
I re-wrote the QueryResult processing so that it more closely resembles what is being done in the iOS version.  I also added skip and limit testing since skip and limit are now implemented.  Done in 3f1b4c2.  

https://github.com/cloudant/sync-android/commit/3f1b4c2992cab575b6d999ec536957bb10d04f57#diff-5f5fe154c6c97af9fc67a7b9f8ae9c87L31
https://github.com/cloudant/sync-android/commit/3f1b4c2992cab575b6d999ec536957bb10d04f57#diff-1a5092ac6698384ba572368c278436fdR311